addons/binding/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderLocalWatcherHandler.java

```diff
@@ -0,0 +1,168 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.folderwatcher.internal.common.WatcherCommon;
+import org.openhab.binding.folderwatcher.internal.config.FolderLocalWatcherConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FolderLocalWatcherHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexandr Salamatov - Initial contribution
+ */
+@NonNullByDefault
+public class FolderLocalWatcherHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FolderLocalWatcherHandler.class);
+
+    @Nullable
+    private FolderLocalWatcherConfiguration config;
+    @Nullable
+    private File currentLocalListingFile;
+    @Nullable
+    private ScheduledFuture<?> executionJob;
+    private ArrayList<String> currentLocalListing = new ArrayList<String>();
+    private ArrayList<String> previousLocalListing = new ArrayList<String>();
+
+    public FolderLocalWatcherHandler(Thing thing) {
+        super(thing);
+
+        currentLocalListingFile = new File(ConfigConstants.getUserDataFolder() + File.separator + "FolderWatcher"
+                + File.separator + thing.getUID().getAsString().replace(':', '_') + ".data");
+
+        try {
+            previousLocalListing = WatcherCommon.initStorage(currentLocalListingFile);
+        } catch (IOException e) {
+            logger.debug("Can't write file: {}", currentLocalListingFile, e);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand for channel {}: {}", channelUID.getId(), command.toString());
+        logger.debug("Command sending not supported by this binding");
+    }
+
+    @Override
+    public void initialize() {
+
+        config = getConfigAs(FolderLocalWatcherConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (config.pollIntervalLocal > 0) {
+            updateStatus(ThingStatus.ONLINE);
+            executionJob = scheduler.scheduleWithFixedDelay(periodicExecutionRunnable, config.pollIntervalLocal,
+                    config.pollIntervalLocal, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval can't be null or negative");
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        executionJob.cancel(true);
+    }
+
+    ArrayList<String> listFiles(String path, boolean listHidden, boolean recursive) throws IOException {
+
+        File[] fileList;
+        ArrayList<String> dirList = new ArrayList<String>();
+
+        // create new file
```

Comment:
Don't comment the obvious. remove those single line comments and add a javadoc to the method instead and explain what is going on here (as this is a central method anyway)